src/suggestion.cpp

```diff
@@ -0,0 +1,325 @@
+  try {
+      database = Xapian::Database(databasefd.release());
+  } catch( Xapian::DatabaseError& e) {
+      std::cerr << "Something went wrong opening xapian database for zimfile "
+                << accessInfo.first << std::endl;
+      std::cerr << "dbOffest = " << accessInfo.second << std::endl;
+      std::cerr << "error = " << e.get_msg() << std::endl;
+      return;
+  }
+
+  m_valuesmap = read_valuesmap(database.get_metadata("valuesmap"));
+  auto language = database.get_metadata("language");
+  if (language.empty() ) {
+      // Database created before 2017/03 has no language metadata.
+      // However, term were stemmed anyway and we need to stem our
+      // search query the same the database was created.
+      // So we need a language, let's use the one of the zim.
+      // If zimfile has no language metadata, we can't do lot more here :/
+      try {
+          language = archive.getMetadata("Language");
+      } catch(...) {}
+  }
+  if (!language.empty()) {
+      icu::Locale languageLocale(language.c_str());
+      /* Configuring language base steemming */
+      try {
+          m_stemmer = Xapian::Stem(languageLocale.getLanguage());
+          m_queryParser.set_stemmer(m_stemmer);
+          m_queryParser.set_stemming_strategy(Xapian::QueryParser::STEM_SOME);
+      } catch (...) {
+          std::cout << "No stemming for language '" << languageLocale.getLanguage() << "'" << std::endl;
+      }
+  }
+
+  m_database.add_database(database);
+}
+
+bool SuggestionDataBase::hasDatabase() const
+{
+  return !m_database.internal.empty();
+}
+
+bool SuggestionDataBase::hasValuesmap() const
+{
+  return !m_valuesmap.empty();
+}
+
+bool SuggestionDataBase::hasValue(const std::string& valueName) const
+{
+  return (m_valuesmap.find(valueName) != m_valuesmap.end());
+}
+
+int SuggestionDataBase::valueSlot(const std::string& valueName) const
+{
+  return m_valuesmap.at(valueName);
+}
+
+/*
+ * subquery_phrase: selects documents that have the terms in the order of the query
+ * within a specified window.
+ * subquery_anchored: selects documents that have the terms in the order of the
+ * query within a specified window and starts from the beginning of the document.
+ * subquery_and: selects documents that have all the terms in the query.
+ *
+ * subquery_phrase and subquery_anchored by themselves are quite exclusive. To
+ * include more "similar" docs, we combine them with subquery_and using OP_OR
+ * operator. If a particular document has a weight of A in subquery_and and B
+ * in subquery_phrase and C in subquery_anchored, the net weight of that document
+ * becomes A+B+C (normalised out of 100). So the documents closer to the query
+ * gets a higher relevance.
+ */
+Xapian::Query SuggestionDataBase::parseQuery(const SuggestionQuery& query)
+{
+  Xapian::Query xquery;
+
+  auto flags = Xapian::QueryParser::FLAG_DEFAULT | Xapian::QueryParser::FLAG_PARTIAL;
+  xquery = m_queryParser.parse_query(query.m_query, flags);
```

Comment:
Should we parse the query using the "S" prefix ?

Reply:
We fall back to `findByTitle` if we do not have a title index. So the prefix have been removed.